lambdas/reset-identity/src/main/java/uk/gov/di/ipv/core/resetidentity/ResetIdentityHandler.java

```diff
@@ -112,17 +113,26 @@ public Map<String, Object> handleRequest(ProcessRequest event, Context context)
 
             // Make sure we do this before deleting the credentials!
             String userName = null;
-            if (isUserInitiated) {
+            CriResponseItem f2fRequest = criResponseService.getFaceToFaceRequest(userId);
+
+            if (isUserInitiated && f2fRequest != null) {
                 List<VcStoreItem> credentials =
                         verifiableCredentialService.getVcStoreItems(
                                 clientOAuthSessionItem.getUserId());
-                userName = getUserName(credentials);
+                userName = getUserName(credentials, false);
+            }
+
+            if (isUserInitiated && f2fRequest == null) {
```

Comment:
Once you update `getUserName()` there will be no difference here between F2F and non-F2F so these blocks can be combined.